src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs

```diff
@@ -275,7 +275,7 @@ protected static T Cast<T>(object value)
             foreach (var member in members)
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                currentDestination = UpdateDestination(availableIndices, options, currentDestination, member, cancellationToken);
+                currentDestination = UpdateDestination(availableIndices, options, currentDestination, member, destination.SyntaxTree.Options, cancellationToken);
```

Comment:
When the symbol is passed to Property/Field/Method Generator, the generated syntax node doesn't have parent. 
So its parse definition would be https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/CSharpParseOptions.cs,21

And later the default option is translated to the latest C# version https://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/LanguageVersion.cs,426
Therefore when this loop is being called, its parsing option may change after the first iteration. (As this is the reason for this bug, C# 5 becomes C# 9 at this point)

The best way I feel here is to first generate all the syntax nodes for members using the same parse option, and then doing the insertion.

This would need to refactor a lot of code in the CodeGenerationService and the method/field/.. generator.
before I do this
tag @CyrusNajmabadi to see if this is the right way to fix this

Reply:
> The best way I feel here is to first generate all the syntax nodes for members using the same parse option, and then doing the insertion.

I'm not opposed to that. However, I feel like something is still wrong here:

> And later the default option is translated to the latest C# version

There should be some way to avoid this. Maybe @jasonmalinowski knows?  It seems patently weird that we fork nodes and trees, but end up changing the options here. Would prefer we avoid that, or force the options to be what we started with.

Reply:
I'm fine with us exploring both options in parallel btw.

Reply:
So the problem is a node created in isolation of a tree doesn't have any options. We create a tree on the fly when asked, but since there's no options it has no choice but to just use the default ones.

The alternative here would be in the path where we have a node and want the option to be remembered would be to explicitly call [CSharpSyntaxTree.Create](https://github.com/dotnet/roslyn/blob/fa5a5bbfc024f0b45a73964f9587d54e28687f30/src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.cs#L313) passing in the random node that we created with the option, and then grab the new red node root back from that. That means later calls to node.Tree.Options will have what you want.